retis/src/core/events/bpf.rs

```diff
@@ -590,8 +591,41 @@ impl FactoryId {
     }
 }
 
-/// Type alias to refer to the commonly used EventSectionFactory HashMap.
-pub(crate) type SectionFactories = HashMap<FactoryId, Box<dyn EventSectionFactory>>;
+/// Map of Section Factories.
+#[derive(Default)]
+pub(crate) struct SectionFactories(HashMap<FactoryId, Box<dyn EventSectionFactory>>);
+
+impl SectionFactories {
+    /// Create a new factory map.
+    pub(crate) fn new() -> Self {
+        Self::default()
+    }
+
+    /// Check whether the map is empty.
+    pub(crate) fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+
+    /// Insert a new factory.
+    pub(crate) fn insert(&mut self, id: FactoryId, factory: Box<dyn EventSectionFactory>) {
+        self.0.insert(id, factory);
+    }
```

Comment:
Suggestion only: if you implement the Deref trait those come for free, in addition to all the other inner methods.